Code/Framework/AzFramework/AzFramework/Physics/HeightfieldProviderBus.h

```diff
@@ -115,9 +118,19 @@ namespace Physics
     public:
         static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Multiple;
 
+        enum class HeightfieldChangeMask
+        {
+            None = 0b00000000,
+            Settings = 0b00000001,
+            HeightData = 0b00000010,
+            MaterialData = 0b00000100,
+            SurfaceData = 0b00001000
+        };
```

Comment:
I don't think this should be an "enum class" for a bitfield, just an enum. Otherwise, you can't mathematically or (|) multiple flags together without typecasting back and forth, which all sort of defeats the purpose of making it a class.

Reply:
(As a side note, it's _possible_ to implement it as enum class with templated operators, and it would be kinda neat to find a way to make this simple and generic so that we can define safe enum bitfields in o3de, but we don't have it yet. See https://accu.org/journals/overload/24/132/williams_2228/ as an example.)

Reply:
We have the [AZ_DEFINE_ENUM_BITWISE_OPERATORS](https://github.com/o3de/o3de/blob/development/Code%2FFramework%2FAzCore%2FAzCore%2Fbase.h#L255-L269) macro in AzCore/base.h

Reply:
That's awesome, thanks for pointing that out!!

Reply:
AZ_DEFINE_ENUM_BITWISE_OPERATORS looks awesome! 
Will look at changing `SceneQuery::HitFlags` and `SceneQuery::ResultFlags` to use it instead of manually defining bitwise operators.